Lib/slapdtest/_slapdtest.py

```diff
@@ -512,60 +512,69 @@ def _cli_popen(self, ldapcommand, extra_args=None, ldap_uri=None,
         args += (extra_args or [])
 
         self._log.debug('Run command: %r', ' '.join(args))
-        proc = subprocess.Popen(
-            args, stdin=subprocess.PIPE, stdout=subprocess.PIPE,
+        proc = subprocess.run(
+            args,
+            input=stdin_data,
+            stdout=subprocess.PIPE,
             stderr=subprocess.PIPE
         )
         self._log.debug('stdin_data=%r', stdin_data)
-        stdout_data, stderr_data = proc.communicate(stdin_data)
-        if stdout_data is not None:
-            self._log.debug('stdout_data=%r', stdout_data)
-        if stderr_data is not None:
-            self._log.debug('stderr_data=%r', stderr_data)
-        if proc.wait() != 0:
+        if proc.stdout is not None:
+            self._log.debug('stdout=%s', proc.stdout.decode("utf-8"))
+        if proc.stderr is not None:
+            self._log.debug('stderr=%s', proc.stderr.decode("utf-8"))
+        if proc.returncode != 0:
             raise RuntimeError(
-                '{!r} process failed:\n{!r}\n{!r}'.format(
-                    args, stdout_data, stderr_data
-                )
+                'Process failed: {!r}'.format(" ".join(args))
```

Comment:
I removed `stdout` and `stderr` here as they have been logged a few lines ago, and large outputs as exceptions messages are totally unpracticable.